test/GithubApi.Gist.test.js

```diff
@@ -0,0 +1,69 @@
+const { StatusCodes } = require('http-status-codes');
+const { expect } = require('chai');
+const axios = require('axios');
+const chai = require('chai');
+const chaiSubset = require('chai-subset');
+
+chai.use(chaiSubset);
+
+const userUrl = 'https://api.github.com/gists';
+const token = process.env.ACCESS_TOKEN;
+const gistUrl = 'https://api.github.com/gists';
+const gist = {
+  description: 'Example of a gist of JS Promise',
+  public: true,
+  files: {
+    'saludoPromesa.js': {
+      content: 'let saludo= new Promise(function(resolve, reject) {\n'
+      + ' const saludar= "Hola Mundo!";\n'
+      + ' setTimeout(() => {\n'
+        + '   if (true){\n'
+          + '     resolve (saludar);\n'
+        + '   }\n'
+        + '   else {\n'
+          + '     reject ("No hemos podido saludarte");\n'
+        + '   }\n'
+      + ' },1000);\n'
+    + '})\n'
+    + 'saludo.then((saludar) => console.log(saludar));\n'
+    + 'saludo.catch(() => console.log(error));\n'
+    + 'saludo.finally(() => console.log("Saludo finalizado"));'
```

Comment:
you can use the backtick ` to define multiline expressions:

```suggestion
      content: `const saludo= new Promise(function(resolve, reject) {
        const saludar= "Hola Mundo!";
        setTimeout(() => {
          if (true) {
            resolve(saludar);
          } else {
            reject("No hemos podido saludarte");
          }
        }, 1000);
      });
      saludo.then((saludar) => console.log(saludar));
      saludo.catch((error) => console.log(error));
      saludo.finally(() => console.log("Saludo finalizado"));`
```